lib/zipkin/json_client.rb

```diff
@@ -38,11 +39,11 @@ def emit_batch(spans)
       request.body = JSON.dump(spans)
       response = http.request(request)
 
-      if response.code != 202
-        STDERR.puts(response.body)
+      if response.code.to_s != '202'
```

Comment:
why are we convering response code to string?

Reply:
Response code is a string in my environment, not sure about the other, so decided to convert it to string before checking.

This was the reason of random new line characters in logs, because the gem logged empty body on successful responses.

Reply:
http://ruby-doc.org/stdlib-2.4.0/libdoc/net/http/rdoc/Net/HTTPResponse.html

> 

The HTTP result code string. For example, ‘302’. You can also determine the response type by examining which response subclass the response object is an instance of.

Looks like you're right. Can you please remove `#to_s` from response.code but lets keep 202 as a string.

Reply:
https://ruby-doc.org/stdlib-2.5.0/libdoc/net/http/rdoc/Net/HTTPResponse.html

The HTTP result code string. For example, '302'. You can also determine the response type by examining which response subclass the response object is an instance of.

Reply:
By the way, I've also seen these blank lines, but didn't have time to look much into it. Thanks for the fix.

Reply:
Removed it.